x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/EsField.java

```diff
@@ -7,33 +7,53 @@
 package org.elasticsearch.xpack.esql.core.type;
 
 import org.elasticsearch.TransportVersions;
-import org.elasticsearch.common.io.stream.NamedWriteable;
-import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.esql.core.util.PlanStreamInput;
+import org.elasticsearch.xpack.esql.core.util.PlanStreamOutput;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 /**
  * Information about a field in an ES index.
  */
-public class EsField implements NamedWriteable {
-    public static List<NamedWriteableRegistry.Entry> getNamedWriteables() {
+public class EsField implements Writeable {
+
+    public record WriteableInfo(String name, Writeable.Reader<? extends EsField> reader) {}
+
+    static final WriteableInfo ENTRY = new WriteableInfo("EsField", EsField::new);
+
+    private static Map<String, WriteableInfo> writeablesMap;
+
+    private static List<WriteableInfo> getWriteablesInfo() {
         return List.of(
             EsField.ENTRY,
             DateEsField.ENTRY,
             InvalidMappedField.ENTRY,
             KeywordEsField.ENTRY,
+            MultiTypeEsField.ENTRY,
             TextEsField.ENTRY,
             UnsupportedEsField.ENTRY
         );
     }
 
-    static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(EsField.class, "EsField", EsField::new);
+    public static WriteableInfo getWriteableInfo(String name) {
+        if (writeablesMap == null) {
+            writeablesMap = getWriteablesInfo().stream().collect(Collectors.toMap(WriteableInfo::name, Function.identity()));
+        }
```

Comment:
I'd build the map in a `static` block if possible. But, I guess, you can't because it's a superclass of some of these? Maybe make an `EsFieldSerialization` for it? I'm not sure, but that's kind of annoying.

Reply:
Yeah, it's a bit of a class hierarchy initialization dance.
Let me simplify it, a map is enough for our needs